src/EditorFeatures/Test2/FindReferences/FindReferencesTests.InternalsVisibleTo.vb

```diff
@@ -0,0 +1,303 @@
+        <Document>
+        [assembly: System.Runtime.CompilerServices.InternalsVisibleTo("P2")]
+
+        public class C
+        {
+            internal static void $${|Definition:Goo|}()
+            {
+            }
+        }
+        </Document>
+    </Project>
+    <Project Language="C#" CommonReferences="true" AssemblyName="P2">
+        <ProjectReference>P1</ProjectReference>
+        <Document>
+        class X
+        {
+            void Bar()
+            {
+                C.[|Goo|]();
+            }
+        }
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WpfTheory, CombinatorialData, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        Public Async Function TestInternalMethodSeenInDownstreamProjectWithIVT_VerbatimString(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true" Name="P1">
+        <Document>
+        using System.Runtime.CompilerServices;
+
+        [assembly: InternalsVisibleTo(@"P2")]
+
+        public class C
+        {
+            internal static void $${|Definition:Goo|}()
+            {
+            }
+        }
+        </Document>
+    </Project>
+    <Project Language="C#" CommonReferences="true" AssemblyName="P2">
+        <ProjectReference>P1</ProjectReference>
+        <Document>
+        class X
+        {
+            void Bar()
+            {
+                C.[|Goo|]();
+            }
+        }
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WpfTheory, CombinatorialData, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        Public Async Function TestInternalMethodSeenInDownstreamProjectWithIVT_UsingAlias(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true" Name="P1">
+        <Document>
+        using X = System.Runtime.CompilerServices.InternalsVisibleToAttribute;
+
+        [assembly: X(@"P2")]
+
+        public class C
+        {
+            internal static void $${|Definition:Goo|}()
+            {
+            }
+        }
+        </Document>
+    </Project>
+    <Project Language="C#" CommonReferences="true" AssemblyName="P2">
+        <ProjectReference>P1</ProjectReference>
+        <Document>
+        class X
+        {
+            void Bar()
+            {
+                C.[|Goo|]();
+            }
+        }
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WpfTheory, CombinatorialData, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        Public Async Function TestInternalMethodSeenInDownstreamProjectWithIVT_NonLiteralString(kind As TestKind, host As TestHost) As Task
```

Comment:
Given we removed the original approach that lead to these tests, do we still need them?

Reply:
Technically no.  But i think ti doesn't hurt.  i.e. if someone came in here later and thought they coudl optimize with syntactic checks, this would catch potential issues.